src/root.tsx

```diff
@@ -10,15 +10,63 @@ import {
     Meta,
     Routes,
     Scripts,
-    Title
+    Title,
+    useNavigate
 } from "solid-start";
 
 // eslint-disable-next-line
 import "@mutinywallet/ui/style.css";
 
+import { App as CapacitorApp } from "@capacitor/app";
+import { Capacitor } from "@capacitor/core";
+import { onCleanup } from "solid-js";
+
 import { ErrorDisplay, I18nProvider, Toaster } from "~/components";
 import { Provider as MegaStoreProvider } from "~/state/megaStore";
 
+function GlobalListeners() {
+    // listeners for native navigation handling
+    // Check if the platform is Android to handle back
+    if (Capacitor.getPlatform() === "android") {
+        const { remove } = CapacitorApp.addListener(
+            "backButton",
+            ({ canGoBack }) => {
+                if (!canGoBack) {
+                    CapacitorApp.exitApp();
+                } else {
+                    window.history.back();
+                }
+            }
+        );
+
+        // Ensure the listener is cleaned up when the component is destroyed
+        onCleanup(() => {
+            console.debug("cleaning up backButton listener");
+            remove();
+        });
+    }
+
+    // Handle app links on native platforms
+    if (Capacitor.isNativePlatform()) {
+        const navigate = useNavigate();
+        const { remove } = CapacitorApp.addListener("appUrlOpen", (data) => {
+            const url = new URL(data.url);
+            const path = url.pathname;
+            const urlParams = new URLSearchParams(url.search);
+
+            console.log(`Navigating to ${path}?${urlParams.toString()}`);
+            navigate(`${path}?${urlParams.toString()}`);
+        });
+
+        onCleanup(() => {
+            console.debug("cleaning up appUrlOpen listener");
+            remove();
+        });
+    }
+
+    return null;
```

Comment:
this is so weird to me to make a null component but seems to work so not gonna fight it!